packages/mobile/e2e/src/utils/twilio.js

```diff
@@ -0,0 +1,42 @@
+import { TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN, VERIFICATION_PHONE_NUMBER } from '@env'
+import twilio from 'twilio'
+import { sleep } from './utils'
+
+const MAX_TRIES = 120
+
+export const receiveSms = async (after = new Date(), numCodes = 3, existingCodes = []) => {
+  try {
+    let client = twilio(TWILIO_ACCOUNT_SID, TWILIO_AUTH_TOKEN)
+    let tryNumber = 0
+
+    while (tryNumber < MAX_TRIES) {
+      const messages = await client.messages.list({
+        to: `${VERIFICATION_PHONE_NUMBER}`,
+        limit: 3,
+        dateSentAfter: after,
+      })
+      const codes = messages.map((message) => message.body.split(': ')[1])
+      // console.log('Codes received:', codes)
+      if (codes.filter((code) => !existingCodes.includes(code)).length >= numCodes) {
+        return codes
```

Comment:
If we return `codes` we are also returning the `existingCodes` again, which is unintended I think. I see this is called with `numCodes = 2` from the resend SMS flow, we probably want to include the filter above right after mapping (line 18).